SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTest.cpp

```diff
@@ -223,59 +287,103 @@ void ASpatialAuthorityTest::PrepareTest()
 			FinishStep();
 		});
 
+		AddStepFromDefinition(NonReplicatedVerifyAuthorityStepDefinition, FWorkerDefinition::Server(1));
+
+		AddStepFromDefinition(NonReplicatedVerifyOthersStepDefinition, FWorkerDefinition::AllWorkers);
+
+		AddStepFromDefinition(NonReplicatedDestroyStepDefinition, FWorkerDefinition::Server(1));
+	}
+
+	// Replicated Dynamic Actor Spawned On Border from Server 2. Server 2 should keep Authority.
+	{
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Border From Server 2 - Spawn"), FWorkerDefinition::Server(2), nullptr, [this]() {
+			ASpatialAuthorityTestReplicatedActor* Actor =
+				GetWorld()->SpawnActor<ASpatialAuthorityTestReplicatedActor>(BorderPosition, FRotator::ZeroRotator);
+			CrossServerSetDynamicReplicatedActor(Actor);
+			FinishStep();
+		});
+
 		AddStep(
-			TEXT("Non-replicated Dynamic Actor - Verify Authority on Server 1"), FWorkerDefinition::Server(1), nullptr, nullptr,
+			TEXT("Replicated Dynamic Actor Spawned On Border From Server 2 - Verify Server 2 Has Authority"), FWorkerDefinition::AllWorkers,
+			nullptr, nullptr,
 			[this](float DeltaTime) {
-				// Not replicated so OnAuthorityGained() is not called.
-				if (VerifyTestActor(DynamicNonReplicatedActor, 1, 1, 0, 0))
+				Timer -= DeltaTime;
```

Comment:
Could you add a comment that we specifically looking for the actor to not switch between servers in this test case?

Reply:
Sure, done